src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -173,43 +173,51 @@ private boolean isFewBlocksAfterInitialization(Task task) {
             String enclaveChallenge = smsService.getEnclaveChallenge(chainTaskId, task.isTeeNeeded());
             if (task.isTeeNeeded() && enclaveChallenge.isEmpty()) continue;
 
-            Optional<RecoveryAction> oRecoveryAction = getAppropriateRecoveryAction(task, replicate, blockNumber);
-            if (!oRecoveryAction.isPresent()) continue;
+            Optional<TaskNotificationType> optionalTaskNotificationType = getAppropriateRecoveryAction(task, replicate, blockNumber);
+            if (!optionalTaskNotificationType.isPresent()) continue;
 
             // generate contribution authorization
             ContributionAuthorization authorization = signatureService.createAuthorization(
```

Comment:
Shouldn't we generate the contribution authorization only in the case of please_contribute notification?

Reply:
Didn't want to do it (-> https://github.com/iExecBlockchainComputing/iexec-core/blob/dc24414a9426e1c0284b57ac2ae4197f6581f1d2/src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java#L183) but done now :P

Reply:
Ah sorry didn't understand you meant that, thanks!